language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -104,43 +104,68 @@
 import org.sonarsource.sonarlint.core.client.api.common.RuleDetails;
 import org.sonarsource.sonarlint.core.client.api.common.analysis.AnalysisResults;
 import org.sonarsource.sonarlint.core.client.api.common.analysis.ClientInputFile;
 import org.sonarsource.sonarlint.core.client.api.common.analysis.Issue;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.IssueListener;
+import org.sonarsource.sonarlint.core.client.api.connected.ConnectedAnalysisConfiguration;
+import org.sonarsource.sonarlint.core.client.api.connected.ConnectedSonarLintEngine;
+import org.sonarsource.sonarlint.core.client.api.connected.ServerConfiguration;
+import org.sonarsource.sonarlint.core.client.api.exceptions.GlobalUpdateRequiredException;
+import org.sonarsource.sonarlint.core.client.api.exceptions.StorageException;
 import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneAnalysisConfiguration;
-import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneGlobalConfiguration;
-import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneGlobalConfiguration.Builder;
 import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneSonarLintEngine;
 import org.sonarsource.sonarlint.core.telemetry.TelemetryPathManager;
 
+import static org.apache.commons.lang.StringUtils.isBlank;
+
 public class SonarLintLanguageServer implements LanguageServer, WorkspaceService, TextDocumentService {
 
+  private static final String USER_AGENT = "SonarLint Language Server";
+
   static final String DISABLE_TELEMETRY = "disableTelemetry";
   static final String TYPESCRIPT_LOCATION = "typeScriptLocation";
   static final String TEST_FILE_PATTERN = "testFilePattern";
-  static final String ANALYZER_PROPERTIES = "analyzerProperties";
+  private static final String ANALYZER_PROPERTIES = "analyzerProperties";
+  static final String CONNECTED_MODE_SERVERS_PROP = "connectedModeServers";
+  static final String CONNECTED_MODE_PROJECT_PROP = "connectedModeProject";
+  private static final String TYPESCRIPT_PATH_PROP = "sonar.typescript.internal.typescriptLocation";
+
   private static final String SONARLINT_CONFIGURATION_NAMESPACE = "sonarlint";
   private static final String SONARLINT_SOURCE = SONARLINT_CONFIGURATION_NAMESPACE;
   private static final String SONARLINT_OPEN_RULE_DESCRIPTION_COMMAND = "SonarLint.OpenRuleDesc";
+  static final String SONARLINT_UPDATE_SERVER_STORAGE_COMMAND = "SonarLint.UpdateServerStorage";
+  static final String SONARLINT_UPDATE_PROJECT_BINDING_COMMAND = "SonarLint.UpdateProjectBinding";
+  private static final List<String> SONARLINT_COMMANDS = Arrays.asList(
+    SONARLINT_OPEN_RULE_DESCRIPTION_COMMAND,
+    SONARLINT_UPDATE_SERVER_STORAGE_COMMAND,
+    SONARLINT_UPDATE_PROJECT_BINDING_COMMAND);
 
   private final SonarLintLanguageClient client;
   private final Future<?> backgroundProcess;
   private final LanguageClientLogOutput logOutput;
+  private final Logger logger;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". ([squid:S1312](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1312))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.core%3Asonarlint-core-parent&pullRequest=192&issues=AWND8neO6NykrOWnW1BV&open=AWND8neO6NykrOWnW1BV)

[ISSUE_KEY]: AWND8neO6NykrOWnW1BV